sma/integrationpluginsma.cpp

```diff
@@ -823,6 +819,24 @@ void IntegrationPluginSma::setupModbusInverterConnection(ThingSetupInfo *info)
     connection->connectDevice();
 }
 
+SpeedwireInterface *IntegrationPluginSma::getSpeedwireInterface()
+{
+    if (m_speedwireInterface && m_speedwireInterface->available())
+        return m_speedwireInterface;
+
+    if (m_speedwireInterface) {
+        delete m_speedwireInterface;
```

Comment:
That's rather dangerous afaict:

I can't spot a direct practical issue right now, given the error and state change handling is not implemented in SpeedWireInterface (they just produce a warning message, which may be an issue on its own, no?). I can see someone fixing that code to handle the error signal at some point (e.g. because during runtime the network interface goes down or whatnot). That most likely would cause m_available to be set to false. This in turn would cause this code to delete the old interface and create a new one, which again would cause things referring to the old one to crash on it. When sharing objects, you shouldn't just delete them without proper handling of the consumers for it (or better, don't destroy it but bring the existing one into an "available" state again).

Reply:
the idea was to re-create it if the interface is not available it will be created again, re-joining the multicast group ecc...

There is currently only one situation making this not available, which is either not binding to the speedwire port or failing in joining the mutlicast group. In both cases a re-creation of the socket might work.

Reply:
but you are right, I need to fix this, all connection underneesth would break...